### PR TITLE
New spec "ipcs -s -i xxx"

### PR DESCRIPTION
**Pull request by huali027**
Signed-off-by: Huanhuan Li <huali@redhat.com>

### All Pull Requests:

Check all that apply:

* [ x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:
Add the spec back since it is used by an existing webserver rule.

### CODE REVIEW

**Review by xiangce — COMMENTED** [review]
Better remove the verb `get_` from the filename and the name of the spec, and use `ipcs` as the filename, `semid` as the name of spec.

File: insights/specs/datasources/get_semid.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from insights.core.context import HostContext
```
Comment (xiangce) [inline insights/specs/datasources/get_semid.py:1]: Please add a title for this module

File: insights/specs/datasources/get_semid.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+from insights.core.plugins import datasource
+from insights.parsers.ipcs import IpcsS
```
Comment (xiangce) [inline insights/specs/datasources/get_semid.py:3]: If there is possible to get the expected list from an existing spec, do not use the Parser which is heavier.

The existing spec `ipcs_s` is quite simple, and can be used directly for this datasource.

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/collect.py [hunk 3]
```diff
@@ -180,6 +180,9 @@
           enabled: true
         - name: insights.components.rhel_version.IsRhel8
           enabled: true
+    # neededfor the 'pre-check' of 'ipcs_s_i' spec
```
Comment (xiangce) [inline insights/collect.py:183]: By change to base on the `ipcs_s` spec, this pre-check parser is not required.

File: insights/specs/datasources/ipcs.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    """
+    content = broker[LocalSpecs.ipcs_s_cmd].content
+    results = set()
+    for s in content:
+        s_splits = s.split()
+        # key        semid      owner      perms      nsems
+        # 0x00000000 65536      apache     600        1
+        if len(s_splits) == 5 and s_splits[1].isdigit():
+            results.add(s_splits[1])
+    return list(results)
```
Comment (xiangce) [inline insights/specs/datasources/ipcs.py:44]: How about skipping it when nothing is get from the `ipcs -s`?

**Review by xiangce — APPROVED** [review]
Looks good. But it would be better if the datasource `semid` could handle the empty result itself.

Approved pending the approval of spec.

**Comment by huali027** [comment]
> Looks good. But it would be better if the datasource `semid` could handle the empty result itself.
> 
> Approved pending the approval of spec.

Sure, I'll check if it's emtpy, thanks for the review.

**Review by xiangce — APPROVED** [review]
Approved pending the approval of Spec

File: insights/specs/default.py [hunk 5]
```diff
@@ -448,7 +448,8 @@ def httpd_cmd(broker):
     ipaupgrade_log = simple_file("/var/log/ipaupgrade.log")
     ipcs_m = simple_command("/usr/bin/ipcs -m")
     ipcs_m_p = simple_command("/usr/bin/ipcs -m -p")
-    ipcs_s = simple_command("/usr/bin/ipcs -s")
+    ipcs_s = ipcs.LocalSpecs.ipcs_s_cmd
```
Comment (bfahr) [inline insights/specs/default.py:451]: @huali027 this spec can be left in the original form as a `simple_command`.  There's no need to create the LocalSpec in the datasource.

File: insights/specs/datasources/ipcs.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+from insights.core.spec_factory import simple_command
+from insights.core.plugins import datasource
+from insights.specs import Specs
+from insights.core.dr import SkipComponent
+
+
+class LocalSpecs(Specs):
+    """ Local specs used only by semid datasources """
+
+    ipcs_s_cmd = simple_command("/usr/bin/ipcs -s")
```
Comment (bfahr) [inline insights/specs/datasources/ipcs.py:15]: There's no need to create this, just reference the spec `insights.specs.Specs.ipcs_s` instead.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/specs/datasources/ipcs.py (hunk 6 above)
Comment (huali027) [reply insights/specs/datasources/ipcs.py]: @bfahr, sure, updated, thanks.

**Review by bfahr — APPROVED** [review]

**Pull request merged**